client/components/Condition/Condition.js

```diff
@@ -1,26 +1,38 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import Input from 'components/Input/Input';
 import styles from './Condition.css';
+import PriceSelector from '../PriceSelector/PriceSelector';
 
-export default class Condition extends PureComponent {
-  static propTypes = {
-    condition: PropTypes.object,
-    action: PropTypes.func,
-  };
-
-  handleOnBlurAction = (e) => {
-    this.props.action(e.target.value);
-  }
-
-  render() {
-    const { condition: { radius } } = this.props;
-    return (
-      <div className={styles.root}>
-        <span>radius:</span>
-        <Input defaultValue={radius} onBlurAction={this.handleOnBlurAction}></Input>
+const Condition = ({ condition: { price, radius }, updater }) => (
+  <div className={styles.root}>
+    <div className={styles.radiusContainer}>
+      <label>Radius</label>
+      <div className={styles.radiusInputContainer}>
+        <Input
+          defaultValue={radius}
+          onBlurAction={e => updater('radius', e.target.value)}
+        />
         <span>meters</span>
       </div>
-    );
-  }
-}
+    </div>
+    <PriceSelector value={price} onChange={v => updater('price', v)} />
+  </div>
+);
+
+Condition.propTypes = {
+  condition: PropTypes.shape({
+    radius: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    price: PropTypes.object,
+  }),
+  updater: PropTypes.func.isRequired,
+};
+
+Condition.defaultProps = {
+  condition: {
+    radius: 500,
```

Comment:
i dont think setting a default radius value is a component's duty...

Reply:
Alright.